tools/tpm2_nvread.c

```diff
@@ -52,12 +55,18 @@ struct tpm_nvread_ctx {
     UINT32 offset;
     TPMS_AUTH_COMMAND session_data;
     char *output_file;
+    char *raw_pcrs_file;
+    SESSION *policy_session;
+    TPML_PCR_SELECTION pcr_selection;
+    struct {
+        UINT8 L : 1;
+        UINT8 F : 1;
```

Comment:
I don't think the F flag is needed, you can just check for raw_pcrs_file being NULL (although the check is not needed as explained in my other comment).

I also wonder if the L flag is needed, maybe you can check for pcr_selection.count != 0 ?

Reply:
I thought about that, but I remember you saying you liked seeing if (flag.x) { do something } vs if (!file) { do something }

Reply:
@williamcroberts I don't remember mentioning that, but I don't have a strong opinion either.